### ISSUE TITLE
Improvement of to_index and to_word

### ISSUE THREAD

**Issue by WuDan0399** [fastnlp/fastNLP#126]
When a word index Vocabulary is build, some of the words in test set are not available in this vocabulary and I choose to use a new parameter `use_unknown` to control. If we want to the index of '<UNK>' for a word not in this Vocabulary,  set `self.use_unknown=True` and  Vocabulary.to_index will return the index of '<UNK>'.

**Comment by FengZiYjun** [comment]
@WuDan0399 
>  If we want to the index of '' for a word not in this Vocabulary, set self.use_unknown=True and Vocabulary.to_index will return the index of ''.

Sorry, I am not clear about what you mean here. But I guess you are talking about indices of words that appear in training set but not in test set. 

`Vocabulary` has a parameter `unknown` to set the tag of unknown token.
https://github.com/fastnlp/fastNLP/blob/3219b9da33065035a05762458335c511b1b8b130/fastNLP/core/vocabulary.py#L50

Here, the unknown token is assigned with default index 1.
https://github.com/fastnlp/fastNLP/blob/3219b9da33065035a05762458335c511b1b8b130/fastNLP/core/vocabulary.py#L102

If you want to change the index assigned to `<unk>`, there is possibly no other way to do that except c hanging the source code yourself.

**Comment by WuDan0399** [comment]
I am talking about indices of words that appear in test set but not in train set.

**Comment by FengZiYjun** [comment]
That's what I mean.  If any word does not appear in the vocabulary, its index is `vocab.to_index('<unk>')`, which is `1`.
